cranelift/filetests/src/test_run.rs

```diff
@@ -115,6 +116,35 @@ impl SubTest for TestRun {
     }
 
     fn run(&self, func: Cow<ir::Function>, context: &Context) -> anyhow::Result<()> {
+        // If this test requests to run on a completely different
+        // architecture than the host platform then we skip it entirely,
+        // since we won't be able to natively execute machine code.
+        let requested_arch = context.isa.unwrap().triple().architecture;
+        #[cfg(target_arch = "riscv64")]
+        {
+            use target_lexicon::Riscv64Architecture;
+            match (requested_arch, Architecture::host()) {
+                // riscv64 can run on riscv64 and riscv64gc and ...
+                // we cannot use an simple requested_arch != Architecture::host() to decide.
```

Comment:
I think this might be an old comment, we now not only skip based on arch but check the native ISA flags  to see if the host can run the particular test.

Reply:
ok,I have merged the newest code.